src/main.rs

```diff
@@ -19,31 +20,113 @@ use directories_next::ProjectDirs;
 #[cfg(feature = "systray")]
 mod tray_icon;
 
-#[derive(Clone)]
-struct Account {
+#[derive(Deserialize)]
+struct Config {
+    #[cfg(feature = "systray")]
+    icons: Option<tray_icon::Icons>,
+    #[serde(rename = "account")]
+    accounts: Vec<ConfigAccount>,
+}
+
+impl Config {
+    fn from_file<P: AsRef<Path>>(path: P) -> anyhow::Result<Self> {
+        let mut file = File::open(path).context("Could not open configuration file at: {path}")?;
+        let mut config_contents = String::new();
+
+        file.read_to_string(&mut config_contents)
+            .context("Could not read configuration file")?;
+
+        toml::from_str(&config_contents).context("Could not parse configuration")
+    }
+}
+
+/// A representation of an account as is available in the configuration
+#[derive(Clone, Deserialize)]
+struct ConfigAccount {
     name: String,
-    server: (String, u16),
+    server: String,
+    port: u16,
     username: String,
-    password: String,
+    password: Option<String>,
     notification_command: Option<String>,
+    folder: Option<String>,
+    #[serde(default)]
     folders: Vec<String>,
+    pwcmd: Option<String>,
 }
 
-#[derive(Clone)]
-struct AccountFolder {
-    account: Arc<Account>,
+/// A representation of an account as is used by Buzz.
+///
+/// This version of the accounts is sanitized and is duplicated into a
+/// separate account for every folder.
+#[derive(Clone, Deserialize, Debug)]
+struct ConnectionAccount {
+    name: String,
+    server: String,
+    port: u16,
+    username: String,
+    password: String,
+    notification_command: Option<String>,
     folder: String,
 }
 
-impl AccountFolder {
+impl ConfigAccount {
+    fn to_connection_accounts(config: &ConfigAccount) -> anyhow::Result<Vec<ConnectionAccount>> {
```

Comment:
```suggestion
    fn to_connection_accounts(self) -> anyhow::Result<Vec<ConnectionAccount>> {
```
that way you can also avoid a couple of the clones inside of here I think

Reply:
The suggestion is fine and works, but I don't think it will remove any clones, as the clones are inherently needed because we go from `Account` to multiple `AccountFolders`. So all account related fields need to be cloned, or switched to some sort of reference.

Reply:
Also renamed the function to use the `into` prefix as recommended by clippy: <https://rust-lang.github.io/rust-clippy/master/index.html#/wrong_self_convention>